pipeline-templates/job/deploy.yml

```diff
@@ -34,6 +34,7 @@ jobs:
             NServiceBusLicense: $(NServiceBusLicense)
             AZURE_CLIENT_SECRET: $(AppRegistrationTemporaryClientSecret)
             EarningsFunctionKey: $(FundingApprenticeshipEarningsFunctionAppSystemKey)
+            EnvironmentName: $(ResourceEnvironmentName)
```

Comment:
This mapping is only needed for secret environment variables.

The code can reference the ResourceEnvironmentName environment variable if needed.